frontend/src/metabase/nav/containers/MainNavbar/BookmarkList/BookmarkList.tsx

```diff
@@ -52,6 +52,20 @@ interface BookmarkItemProps {
 const BOOKMARKS_INITIALLY_VISIBLE =
   localStorage.getItem("shouldDisplayBookmarks") !== "false";
 
+function isBookmarkSelected(bookmark: Bookmark, selectedItem?: SelectedItem) {
+  if (!selectedItem) {
+    return false;
+  }
+  if (isDataAppBookmark(bookmark)) {
+    return (
+      selectedItem.type === "data-app" && selectedItem.id === bookmark.app_id
```

Comment:
Why we didn't reuse `item_id` but used `app_id`? If we used `item_id` there would be no special case for data apps.

Reply:
@ranquild `item_id` in this case is a collection ID and not an app ID that we need. We're building apps on top of collections. Each app has a linked collection in a 1:1 way. We don't display those "app collections" in Metabase as other collections. Apps rely on them under the hood to inherit collection-like features (containing stuff, bookmarks, etc.). So in this case we're actually bookmarking a collection and `app_id` allows us to distinguish it from a regular collection (in a similar way how models bookmarking works — same endpoints and schema as for cards, but we also hydrate `dataset: true` property to identify models)